receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -281,7 +282,12 @@ func (r *pReceiver) initPrometheusComponents(ctx context.Context, host component
 	if err != nil {
 		return err
 	}
-	r.scrapeManager = scrape.NewManager(&scrape.Options{PassMetadataInContext: true}, logger, store)
+	r.scrapeManager = scrape.NewManager(&scrape.Options{
```

Comment:
Is there any means we can test this within the project?

Reply:
I was hoping to avoid duplicating most of testComponent to test this

Reply:
I've added a test

Reply:
I wanted to make sure that this doesn't regress again in future.